pkg/graveler/committed/apply_test.go

```diff
@@ -93,24 +89,22 @@ func TestApplyDelete(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
-	range2 := &committed.Range{ID: "two", MaxKey: committed.Key("dz")}
 	source := testutil.NewFakeIterator()
 	source.
-		AddRange(&committed.Range{ID: "one", MaxKey: committed.Key("cz")}).
+		AddRange(&committed.Range{ID: "one", MinKey: committed.Key("a"), MaxKey: committed.Key("cz"), Count: 3, EstimatedSize: 2012}).
 		AddValueRecords(makeV("a", "source:a"), makeV("b", "source:b"), makeV("c", "source:c")).
-		AddRange(range2).
+		AddRange(&committed.Range{ID: "two", MinKey: committed.Key("d"), MaxKey: committed.Key("dz"), Count: 1, EstimatedSize: 1011}).
 		AddValueRecords(makeV("d", "source:d")).
-		AddRange(&committed.Range{ID: "three", MaxKey: committed.Key("ez")}).
+		AddRange(&committed.Range{ID: "three", MinKey: committed.Key("ez"), MaxKey: committed.Key("ez"), EstimatedSize: 1011, Count: 1}).
 		AddValueRecords(makeV("e", "source:e"))
-	diffs := testutil.NewValueIteratorFake([]graveler.ValueRecord{
-		*makeTombstoneV("b"),
-		*makeTombstoneV("e"),
-	})
-
+	diffs := testutil.NewFakeIterator()
+	diffs.
+		AddRange(&committed.Range{ID: "diffs:one", MinKey: committed.Key("b"), MaxKey: committed.Key("e"), Count: 2, EstimatedSize: 2012}).
+		AddValueRecords(makeTombstoneV("b"), makeTombstoneV("e"))
 	writer := mock.NewMockMetaRangeWriter(ctrl)
 	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("a", "source:a")))
 	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("c", "source:c")))
-	writer.EXPECT().WriteRange(gomock.Eq(*range2))
+	writer.EXPECT().WriteRange(gomock.Eq(committed.Range{ID: "two", MinKey: committed.Key("d"), MaxKey: committed.Key("dz"), Count: 1, EstimatedSize: 1011}))
```

Comment:
Also add a test for a deleted range here? (i.e. EstimatedSize==0)

In general, this test file can be expanded.